File: 1470-shuffle-the-array/1470-shuffle-the-array.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {
    let ansArr = [];
    let idx = 0;
    let fptr = 0;
    let eptr = n;
    while(idx < nums.length){
        if(idx % 2 === 0){
            ansArr[idx] = nums[fptr];
            idx++;
            fptr++;
        }else{
            ansArr[idx] = nums[eptr];
            idx++;
            eptr++;
        }
    }
    return ansArr;
};

File: 1470-shuffle-the-array/1470-shuffle-the-array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int[] ans = new int[nums.length];
        int idx = 0;
        for(int i=0; i<ans.length; i++){
            if(i % 2 == 0){
                ans[i] = nums[idx++];
            }
        }
        for(int i=0; i<ans.length; i++){
            if(i % 2 != 0){
                ans[i] = nums[idx++];
            }
        }
        return ans;
    }
}